### PR TITLE
Add zen-engine package

### PR DESCRIPTION
**Pull request by carinapeng**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by carinapeng** [comment]
@conda-forge/help-python, ready for review!

File: recipes/zen-engine/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+about:
+  home: https://gorules.io/
+  summary: Open-Source Business Rules Engine
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - carinapeng
+    - janjagusch
```
Comment (janjagusch) [inline recipes/zen-engine/meta.yaml:49]: I'd like to co-maintain this package.

File: recipes/zen-engine/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - python                              # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - crossenv                            # [build_platform != target_platform]
+    - maturin >=0.14,<0.15                # [build_platform != target_platform]
```
Comment (janjagusch) [inline recipes/zen-engine/meta.yaml:21]: After this PR gets merged, we need to request an `osx-arm64` package through https://github.com/conda-forge/conda-forge-pinning-feedstock. Example:  https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3435

File: recipes/zen-engine/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - {{ compiler('rust') }}
+  host:      
+    - python
+    - maturin >=0.14,<0.15
+    - pip
+    - rust
+  run:
+    - python
+    - libgcc-ng  # [linux]
```
[the remark this answers is not in the document]
Comment (CarinaPengQC) [reply recipes/zen-engine/meta.yaml:30]: We get an error `conda_build.exceptions.OverLinkingError: overlinking check failed`. One solution suggested by the error is that`zen-engine,lib/python3.11/site-packages/zen/zen.cpython-311-x86_64-linux-gnu.so` shared module has `libgcc-ng` as a dependency, so we need to specify that, and it solved it

**Review by janjagusch — APPROVED** [review]

**Comment by janjagusch** [comment]
@xhochy, can you take a look, please?

**Comment by janjagusch** [comment]
@conda-forge/python, @xhochy, could someone please review this PR? It would be super useful for us to have `zen-engine` on CF.

File: recipes/zen-engine/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - python                              # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - crossenv                            # [build_platform != target_platform]
+    - maturin >=0.14,<0.15                # [build_platform != target_platform]
+    - {{ compiler('rust') }}
```
Comment (ocefpaf) [inline recipes/zen-engine/meta.yaml:22]: This should fix https://github.com/conda-forge/staged-recipes/pull/23504/files#r1285448036

```suggestion
    - {{ compiler('c') }}
    - {{ compiler('rust') }}
```

File: recipes/zen-engine/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/zen-engine/meta.yaml:30]: You can probably remove this when you add https://github.com/conda-forge/staged-recipes/pull/23504/files#r1300026487

**Comment by janjagusch** [comment]
Ping, @carinapeng. :)
Could you please incorporate @ocefpaf's suggestions, so we can merge this? Thank you!

File: recipes/zen-engine/meta.yaml (hunk 4 above)
Comment (carinapeng) [reply recipes/zen-engine/meta.yaml:23]: Thank you so much! Would you be able to review again after the change, please? 😄

**Pull request merged**